packages/openid4vc/src/openid4vc-holder/OpenId4VciHolderService.ts

```diff
@@ -367,6 +368,55 @@ export class OpenId4VciHolderService {
     }
   }
 
+  public async refreshAccessToken(agentContext: AgentContext, options: OpenId4VciTokenRefreshOptions) {
+    const oauth2Client = this.getOauth2Client(agentContext)
+
+    const authorizationServerMetadata = getAuthorizationServerMetadataFromList(
+      options.issuerMetadata.authorizationServers,
+      options.issuerMetadata.authorizationServers.find(
+        (server) =>
+          // TODO: add type in @openid4vc/oauth2
+          Array.isArray(server.grant_types_supported) && server.grant_types_supported.includes('refresh_token')
+      )?.issuer ?? options.issuerMetadata.authorizationServers[0].issuer
+    )
+
+    const isDpopSupported = oauth2Client.isDpopSupported({
+      authorizationServerMetadata,
+    })
+
+    const dpop = options.dpop
+      ? await this.getDpopOptions(agentContext, {
+          ...options.dpop,
+          dpopSigningAlgValuesSupported: [options.dpop.alg],
+        })
+      : // We should be careful about this case. It could just be the user didn't correctly
+        // provide the DPoP from the auth response. In which case different DPoP will be used
+        // However it might be that they only use DPoP for the token request (esp in pre-auth case)
+        isDpopSupported.supported
+        ? await this.getDpopOptions(agentContext, {
+            dpopSigningAlgValuesSupported: isDpopSupported.dpopSigningAlgValuesSupported,
+          })
+        : undefined
```

Comment:
When we have an access token already, we already have the dpop key, so i don't
Think we have to generate a new one (servers should verify that you use the same dpop key to request the new access token as the initial one, so you're sure it's still the same client)